torch_geometric/nn/conv/gen_conv.py

```diff
@@ -143,15 +154,29 @@ def __init__(
         if isinstance(in_channels, int):
             in_channels = (in_channels, in_channels)
 
-        channels = [in_channels[0]]
+        if in_channels[0] != out_channels:
+            self.lin_n = Linear(in_channels[0], out_channels, bias=bias)
+
+        if edge_dim is not None and edge_dim != out_channels:
+            self.lin_e = Linear(edge_dim, out_channels, bias=bias)
+
+        if isinstance(self.aggr_module, MultiAggregation):
+            aggr_out_channels = self.aggr_module.get_out_channels(out_channels)
+        else:
+            aggr_out_channels = out_channels
+
+        if aggr_out_channels != out_channels:
+            self.lin_l = Linear(aggr_out_channels, out_channels, bias=bias)
+
+        if in_channels[1] != out_channels:
+            self.lin_r = Linear(in_channels[1], out_channels, bias=bias)
+
+        channels = [out_channels]
         for i in range(num_layers - 1):
             channels.append(out_channels * 2)
         channels.append(out_channels)
         self.mlp = MLP(channels, norm=norm)
```

Comment:
`MLP` also has a `bias` argument, not sure if we want to use it though.

Reply:
Thanks! Good point. But for the linear layers, the biases are set to `False` by default while it is `True` for the MLP. We can also set them to `True` by default if that makes it cleaner.